services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/ResumableFileDownload.java

```diff
@@ -125,6 +143,129 @@ public Optional<Long> totalSizeInBytes() {
+     *
+     * @param cs encoding charset
+     * @return the serialized JSON string
+     */
+    public String toString(Charset cs) {
+        return new String(ResumableFileDownloadSerializer.toJson(this), cs);
+    }
+
+    /**
+     * Returns the serialized JSON data representing this object as an {@link SdkBytes} object.
+     *
+     * @return the serialized JSON as {@link SdkBytes}
+     */
+    public SdkBytes toBytes() {
+        return SdkBytes.fromByteArray(ResumableFileDownloadSerializer.toJson(this));
+    }
+
+    /**
+     * Returns the serialized JSON data representing this object as a {@link ByteArrayInputStream}.
+     *
+     * @return the serialized JSON input stream
+     */
+    public InputStream toInputStream() {
+        return new ByteArrayInputStream(ResumableFileDownloadSerializer.toJson(this));
+    }
+
+    /**
+     * Deserialize data at the given path into a {@link ResumableFileDownload}. The file must be written as
+     * Base64 encoded JSON.
+     *
+     * @param path The {@link Path} to the file with serialized data
+     * @return the deserialized {@link ResumableFileDownload}
+     * @throws IOException if there are issues reading from the file
+     */
+    public static ResumableFileDownload fromFile(Path path) throws IOException {
+        return ResumableFileDownloadSerializer.fromJson(fromBase64Bytes(Files.readAllBytes(path)));
+    }
+
+    /**
+     * Deserialize a ByteBuffer with JSON data into a {@link ResumableFileDownload}.
+     *
+     * @param byteBuffer the serialized data
+     * @return the deserialized {@link ResumableFileDownload}
+     */
+    public static ResumableFileDownload fromByteBuffer(ByteBuffer byteBuffer) {
+        byte[] bytes = new byte[byteBuffer.remaining()];
+        byteBuffer.get(bytes);
+        return ResumableFileDownloadSerializer.fromJson(bytes);
+    }
+
+    /**
+     * Deserialize a byte array with JSON data into a {@link ResumableFileDownload}.
+     *
+     * @param bytes the serialized data
+     * @return the deserialized {@link ResumableFileDownload}
+     */
+    public static ResumableFileDownload fromBytes(byte[] bytes) {
+        return ResumableFileDownloadSerializer.fromJson(bytes);
+    }
+
+    /**
+     * Deserialize contents of an input stream with JSON data into a {@link ResumableFileDownload}.
+     *
+     * @param inputStream the stream containing serialized data
+     * @return the deserialized {@link ResumableFileDownload}
+     */
+    public static ResumableFileDownload fromInputStream(InputStream inputStream) throws IOException {
+        return ResumableFileDownloadSerializer.fromJson(toByteArray(inputStream));
+    }
+
+    /**
+     * Deserialize a string with JSON data into a {@link ResumableFileDownload}.
+     *
+     * @param contents the serialized data
+     * @return the deserialized {@link ResumableFileDownload}
+     */
+    public static ResumableFileDownload fromString(String contents) {
```

Comment:
I just noticed we have `utf8` in the method name else where for this kind of method. Should it be `fromUtf8String`? `toString` would be `toUtf8String`, and then the overridden `toString` would be the standard SDK toString format. WDYT?

https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/IoUtils.java#L58

Reply:
Yes, it makes sense. 

I included a `toString(Charset)` method too, because we have a fromString(Charset) method and thought symmetry was better. Do you think it's inconsistent if it's called `toString`?